gnucash/import-export/import-main-matcher.c

```diff
@@ -1559,49 +1559,47 @@ get_required_color (const gchar *class_name)
+                     const char *account, const char *amount, const char *narrative)
+{
+    GtkTreeStore *store = GTK_TREE_STORE (model);
+    GtkTreeIter child;
+    gtk_tree_store_append (store, &child, iter);
+    gtk_tree_store_set (store, &child, DOWNLOADED_COL_DATE_TXT, date, -1);
+    gtk_tree_store_set (store, &child, DOWNLOADED_COL_ACCOUNT, account, -1);
+    gtk_tree_store_set (store, &child, DOWNLOADED_COL_AMOUNT, amount, -1);
+    gtk_tree_store_set (store, &child, DOWNLOADED_COL_DESCRIPTION, narrative, -1);
+    gtk_tree_store_set (store, &child, DOWNLOADED_COL_ENABLE, FALSE, -1);
+}
+
 static void
 update_child_row (GNCImportMatchInfo *sel_match, GtkTreeModel *model, GtkTreeIter *iter)
 {
-    GtkTreeStore *store;
-    GtkTreeIter  child;
-    gchar *text = qof_print_date (xaccTransGetDate (sel_match->trans));
-    const gchar *ro_text;
-
-    const gchar *desc = xaccTransGetDescription (sel_match->trans);
-    const gchar *memo = xaccSplitGetMemo (sel_match->split);
+    const Transaction *trans = sel_match->trans;
+    gchar *datestr = qof_print_date (xaccTransGetDate (trans));
 
-    store = GTK_TREE_STORE(model);
+    remove_child_row (model, iter);
+    add_row (model, iter, datestr, "", "", xaccTransGetDescription (trans));
+    g_free (datestr);
 
-    if (!gtk_tree_model_iter_has_child (model, iter))
-        gtk_tree_store_append (GTK_TREE_STORE(model), &child, iter);
-    else
-        gtk_tree_model_iter_nth_child (model, &child, iter, 0);
-
-    gtk_tree_store_set (store, &child, DOWNLOADED_COL_DATE_TXT, text, -1);
-
-    if (xaccTransCountSplits (sel_match->trans) == 2)
-        gtk_tree_store_set (store, &child, DOWNLOADED_COL_ACCOUNT, xaccAccountGetName (
-                            xaccSplitGetAccount (xaccSplitGetOtherSplit (sel_match->split))), -1);
-    else
-        gtk_tree_store_set (store, &child, DOWNLOADED_COL_ACCOUNT, _("-- Split Transaction --"), -1);
-
-    ro_text = xaccPrintAmount (xaccSplitGetAmount (sel_match->split),
-                               gnc_split_amount_print_info (sel_match->split, TRUE));
-
-    gtk_tree_store_set (store, &child, DOWNLOADED_COL_AMOUNT, ro_text, -1);
-    gtk_tree_store_set (store, &child, DOWNLOADED_COL_MEMO, memo, -1);
-    gtk_tree_store_set (store, &child, DOWNLOADED_COL_DESCRIPTION, desc, -1);
-
-    gtk_tree_store_set (store, &child, DOWNLOADED_COL_ENABLE, FALSE, -1);
-    g_free (text);
+    for (GList *n = xaccTransGetSplitList (trans); n; n = n->next)
+    {
+        Split *s = n->data;
+        const gchar *acct_name = xaccAccountGetName (xaccSplitGetAccount (s));
+        const gchar *memo = xaccSplitGetMemo (s);
+        const gchar *amount = xaccPrintAmount (xaccSplitGetAmount (s),
+                                               gnc_split_amount_print_info (s, TRUE));
+        add_row (model, iter, (sel_match->split == s) ? "==>" : "",
+                 acct_name, amount, (memo && *memo) ? memo : "(no memo)");
+    }
```

Comment:
I was not sure how to show my changes for this so added here...

    const Transaction *trans = sel_match->trans;
    const gchar *desc = xaccTransGetDescription (trans);
    gchar *datestr = qof_print_date (xaccTransGetDate (trans));

    remove_child_row (model, iter);

    for (GList *n = xaccTransGetSplitList (trans); n; n = n->next)
    {
        GtkTreeStore *store = GTK_TREE_STORE (model);
        GtkTreeIter child;
        Split *s = n->data;
        const gchar *acct_name = xaccAccountGetName (xaccSplitGetAccount (s));
        const gchar *memo = xaccSplitGetMemo (s);
        const gchar *amount = xaccPrintAmount (xaccSplitGetAmount (s),
                                               gnc_split_amount_print_info (s, TRUE));

        gtk_tree_store_append (store, &child, iter);
        gtk_tree_store_set (store, &child,
                            DOWNLOADED_COL_DATE_TXT, (sel_match->split == s) ? datestr : "",
                            DOWNLOADED_COL_ACCOUNT, acct_name,
                            DOWNLOADED_COL_AMOUNT, amount,
                            DOWNLOADED_COL_DESCRIPTION, (sel_match->split == s) ? desc : "",
                            DOWNLOADED_COL_MEMO, (memo && *memo) ? memo : "(no memo)",
                            DOWNLOADED_COL_ENABLE, FALSE, -1);
    }
    g_free (datestr);

You will see that I have added info to the memo column and also populated the existing transaction description and removed the extra date row as the date is now added to the split row.
Try it and see if it works for you.

Reply:
The main reason for the separate function is that I've moved the existing splits' memo into the Description header, similar to the regular register. Otherwise the memos are squashed into a separate column while the description column is empty.

Reply:
I see what you mean, then maybe there should only be one column, combine Description and Memo and change column header appropriately on row selection. May work?
I still would get rid of the extra line though.

Reply:
:+1: for combining Desc and Memo. They could even be combined via `\n` in the imported splits. But it needs to be tested for UX.

The extraline would be needed for the matched transaction - txn->description in 1 line, and 1 line per split->memo.